app/services/email_output_service.rb

```diff
@@ -24,20 +26,44 @@ def generate_attachments(action:, attachments:, pdf_attachment:)
     email_attachments = []
 
     if action.fetch(:include_attachments, false)
-      email_attachments = email_attachments.concat attachments
+      email_attachments = email_attachments.concat(by_size(attachments))
     end
     if action.fetch(:include_pdf, false)
       email_attachments.prepend(pdf_attachment)
     end
 
-    email_attachments
+    attachments_per_email(email_attachments)
+  end
+
+  def by_size(attachments)
+    attachments.sort_by { |attachment| attachment.size }
+  end
+
+  def sum(attachments, to_add)
+    attachments.inject(0) { |sum, attachment| sum + attachment.size } + to_add.size
+  end
+
+  def attachments_per_email(attachments)
+    all = []
+    per_email = []
+
+    attachments.each do |attachment|
+      if sum(per_email, attachment) >= MAX_ATTACHMENTS_SIZE
+        all << per_email
+        attachment == attachments.last ? all << [attachment] : per_email = [attachment]
+      else
+        per_email << attachment
+        all << per_email if attachment == attachments.last
+      end
+    end
+    all
```

Comment:
I don't have any suggestion but did you consider moving this to another class or have a mix of methods with the Attachment object?

Reply:
yeah i can move it all out into its' own class 👍